api/meter.go

```diff
@@ -250,6 +280,50 @@ func (r *relayMeter) AppRelays(app string, from, to time.Time) (AppRelaysRespons
 	return resp, nil
 }
 
+func (r *relayMeter) AppLatency(app string) (AppLatencyResponse, error) {
+	r.Logger.WithFields(logger.Fields{"app": app}).Info("apiserver: Received AppLatency request")
+
+	appLatency := r.todaysLatency[app]
+
+	if len(appLatency) == 0 {
+		return AppLatencyResponse{}, ErrAppLatencyNotFound
+	}
+
+	sort.Slice(appLatency, func(i, j int) bool {
+		return appLatency[i].Time.Before(appLatency[j].Time)
+	})
+
+	return AppLatencyResponse{
+		Application:  app,
+		DailyLatency: appLatency,
+		From:         appLatency[0].Time,
+		To:           appLatency[len(appLatency)-1].Time,
+	}, nil
+}
+
+func (r *relayMeter) AllAppsLatencies() ([]AppLatencyResponse, error) {
+	r.Logger.Info("apiserver: Received AllAppsLatencies request")
+
+	resp := []AppLatencyResponse{}
+
+	for app, appLatency := range r.todaysLatency {
+		sort.Slice(appLatency, func(i, j int) bool {
+			return appLatency[i].Time.Before(appLatency[j].Time)
+		})
+
+		latencyResp := AppLatencyResponse{
+			Application:  app,
+			DailyLatency: appLatency,
+			From:         appLatency[0].Time,
```

Comment:
Is it ever possible, e.g. due to a db error, to have an empty array as `appLatency`? If so, that could cause a panic here.

Reply:
Updated to avoid adding to response if Latency response slice is empty.